catalyst_voices/packages/internal/catalyst_voices_models/lib/src/proposal_schema/proposal_schema.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:catalyst_voices_models/src/proposal_schema/proposal_definitions.dart';
+
+class ProposalSchema {
+  final String schema;
+  final String title;
+  final String description;
+  final ProposalDefinitions definitions;
+  final DefinitionsObjectTypes type;
+  final bool additionalProperties;
+  final List<ProposalSchemaSegment> properties;
+  final List<String> order;
+
+  const ProposalSchema({
+    required this.schema,
+    required this.title,
+    required this.description,
+    required this.definitions,
+    required this.type,
+    required this.additionalProperties,
+    required this.properties,
+    required this.order,
+  });
+}
+
+class ProposalSchemaSegment {
+  final BaseProposalDefinition ref;
+  final String id;
+  final String title;
+  final String description;
+  final List<ProposalSchemaSection> properties;
+  final List<String> required;
+  final List<String> order;
+
+  const ProposalSchemaSegment({
+    required this.ref,
+    required this.id,
+    required this.title,
+    required this.description,
+    required this.properties,
+    required this.required,
+    required this.order,
+  });
+}
+
+class ProposalSchemaSection {
+  final BaseProposalDefinition ref;
+  final String id;
+  final String title;
+  final String description;
+  final List<ProposalSchemaElement> properties;
+  final List<String> required;
+  final List<String> order;
+
+  const ProposalSchemaSection({
+    required this.ref,
+    required this.id,
+    required this.title,
+    required this.description,
+    required this.properties,
+    required this.required,
+    required this.order,
+  });
+}
+
+class ProposalSchemaElement {
+  final BaseProposalDefinition ref;
+  final String id;
+  final String title;
+  final String description;
+  final int? minLength;
+  final int? maxLength;
+  final String? defaultValue;
+  final String guidance;
+  final List<String>? enumValues;
+  final int? maxItems;
+  final int? minItems;
+  final int? minimum;
+  final int? maximum;
```

Comment:
Can we have at the top def like Range? Maybe somewhere in `shared` package it would make more sense.

```dart
typedef Range<T extends num> = ({T min, T max});
typedef IntRange = Range<int>;
```
and here use them? `rangeItems` / `range`;

Reply:
there is no information about if two values (min, max) will be provided or only one. Then `typedef Range<T extends num?> = ({T min, T max});` the num should be nullable